examples/structured_2d_dgsem/elixir_shallowwater_well_balanced_wet_dry.jl

```diff
@@ -0,0 +1,235 @@
+
+###############################################################################
+# Workaround to compute the well-balancedness error for this partiular problem
+# that has two reference water heights. One for a lake to the left of the
+# discontinuous bottom topography `H0_upper = 2.5` and another for a lake to the
+# right of the discontinuous bottom topography `H0_lower = 1.5`.
+
+# Declare a special version of the function to compute the lake-at-rest error
+# OBS! The reference water height values are hardcoded for convenience.
+function lake_at_rest_error_two_level(u, element_id, equations::ShallowWaterEquations2D)
+  h, _, _, b = u
+
+  # For well-balancedness testing with possible wet/dry regions the reference
+  # water height `H0` accounts for the possibility that the bottom topography
+  # can emerge out of the water as well as for the threshold offset to avoid
+  # division by a "hard" zero water heights as well.
+
+  # very brute force approach to recovery the appropriate reference water height
+  # on the StructuredMesh
+  if ((element_id >= 1   && element_id <= 8  ) ||
+      (element_id >= 17  && element_id <= 24 ) ||
+      (element_id >= 33  && element_id <= 40 ) ||
+      (element_id >= 49  && element_id <= 56 ) ||
+      (element_id >= 65  && element_id <= 72 ) ||
+      (element_id >= 81  && element_id <= 88 ) ||
+      (element_id >= 97  && element_id <= 104) ||
+      (element_id >= 113 && element_id <= 120) ||
+      (element_id >= 129 && element_id <= 136) ||
+      (element_id >= 145 && element_id <= 152) ||
+      (element_id >= 161 && element_id <= 168) ||
+      (element_id >= 177 && element_id <= 184) ||
+      (element_id >= 193 && element_id <= 200) ||
+      (element_id >= 209 && element_id <= 216) ||
+      (element_id >= 225 && element_id <= 232) ||
+      (element_id >= 241 && element_id <= 248))
+    H0_wet_dry = max( 2.5 , b + equations.threshold_limiter )
+  else
+    H0_wet_dry = max( 1.5 , b + equations.threshold_limiter )
+  end
+
+  return abs(H0_wet_dry - (h + b))
+end
+
+# point to the data we want to analyze
+u = Trixi.wrap_array(sol[end], semi)
+# Perform the actual integration of the well-balancedness error over the domain
+l1_well_balance_error = Trixi.integrate_via_indices(u, mesh, equations, semi.solver, semi.cache; normalize=true) do u, i, j, element, equations, solver
+  u_local = Trixi.get_node_vars(u, equations, solver, i, j, element)
+  return lake_at_rest_error_two_level(u_local, element, equations)
+end
+
+# report the well-balancedness lake-at-rest error to the screen
+if Trixi.mpi_isroot()
+  Trixi.mpi_println("─"^100)
+  Trixi.mpi_println(" Lake-at-rest error for '", Trixi.get_name(equations), "' with ", Trixi.summary(solver),
+                    " at final time " * Trixi.@sprintf("%10.8e", tspan[end]))
+
+  Trixi.@printf(" %-12s:", Trixi.pretty_form_utf(lake_at_rest_error))
+  Trixi.@printf("  % 10.8e", l1_well_balance_error)
+  Trixi.mpi_println()
+  Trixi.mpi_println("─"^100)
+end
```

Comment:
This setup will not work with MPI for several reasons
- The structured mesh does not support MPI at all
- Even if it supported MPI, the element IDs would be different in parallel

Thus, please change this part so that it doesn't look like it supports MPI and leave some comments explaining this.

Reply:
Okay, because the element IDs would differ should I change the `TreeMesh` version of this output as well? TBH I just copied this output strategy from some of the other analysis routines. Also, it would be very strange to run this test in parallel as it is just to test well-balancedness. It has no physical meaning.

Reply:
Yes, please :+1:

Reply:
Okay, I will revert everything to serial as to not cause confusion / overpromise among users.

Reply:
Done. I removed the mpi specific parts from this output.